src/common/components/header/header.tsx

```diff
@@ -11,12 +11,21 @@ import { Button } from "primereact/button";
 interface HeaderProps {}
 
 export const Header: React.FC<HeaderProps> = ({}) => {
+  const navigate = useNavigate();
+
   const user = JSON.parse(localStorage.getItem("user") || "{}");
+  const [isAuthenticated, setIsAuthenticated] = useState(true);
```

Comment:
I think I saw this state in multiple components and similar storage checks made on it. Since this data is stored in loginStore as well you could update the isAuthenticated prop from there with some actions and use them anywhere they are needed. This way you know that you have the same auth state in the entire app and you can get rid of the multiple isAuthenticated local states.